config_loader.py

```diff
@@ -49,24 +126,43 @@ def reload_config(self):
                 self.__property_dictionary = yaml.load(stream)
             except yaml.YAMLError as e:
                 print(e)
+                raise e
+
+    @property
+    def properties(self):
+        """
+        Get property dictionary
+        :return: Property Dictionary
+        """
 
-    '''
-    Get property dictionary
-    '''
-    def get_properties(self):
         return self.__property_dictionary
 
 
-'''
-Using singleton pattern to maintain one single (global) configuration for the whole program.
-The input parameter 'yaml_file' will be ignored if the configuration object has already been loaded
-'''
+# Global Config DAO available for access via import
+# Should be initialized when this module is imported
+CONFIG: _ConfigDAO = None
```

Comment:
Is this valid syntax? What does it do?

Reply:
Python 3.5+ syntax. Referred to as "type hinting". This is mostly just a IDE feature so that I can use auto complete functionality. Similar to Java annotations in the fact that it can actually catch compiler errors. For example, if I call getDeskName() on a CONFIG object by accident, the IDE should catch this and tell us (without needing to execute that line of code and generate a Python Error b/c the method doesn't exist)

See the image below for an example:
![screen shot 2017-11-18 at 1 16 53 pm](https://user-images.githubusercontent.com/10552190/32983452-c54882b6-cc62-11e7-80a4-d7b33bad9c72.png)

(Taken From http://www.daveoncode.com/2017/03/06/writing-better-software-with-python-3-6-type-hints/)

Reply:
I will slowly try to change all function arguments and variable definitions so that we can utilize this functionality. This adds some extra checking to our variables to ensure that we are passing the right object instances and calling the right methods. This can save a lot of headache down the line if our code base expands.

Reply:
Awesome, thanks for introducing this to me.